test/functional/eval/input_spec.lua

```diff
@@ -439,3 +440,40 @@ describe('inputdialog()', function()
     ]])
   end)
 end)
+
+describe('confirm()', function()
+  it("shows dialog even if 'silent' #8788", function()
+    command("autocmd BufNewFile * call confirm('test')")
+
+    local function check_and_clear(edit_line)
+      screen:expect([[
+                                 |
+        {SEP:                         }|
+        ]]..edit_line..[[
+        {CONFIRM:test}                     |
+        {CONFIRM:[O]k: }^                   |
+      ]])
+      feed('<cr>')
+      command('redraw')
+      command('bdelete!')
+    end
+
+    -- With shortmess-=F
+    command('set shortmess-=F')
+    feed(':edit foo<cr>')
+    check_and_clear('"foo" [New File]         |\n')
+
+    -- With shortmess+=F
+    command('set shortmess+=F')
+    feed(':edit foo<cr>')
+    check_and_clear(':edit foo                |\n')
+
+    -- With :silent
+    feed(':silent edit foo<cr>')
+    check_and_clear(':silent edit foo         |\n')
+
+    -- With API (via eval/VimL) call and shortmess+=F
+    feed(':call nvim_command("edit x")<cr>')
```

Comment:
What about just `command('edit x')` (rpc api), will it work the same? Regardless it could be added to the test.

Reply:
Not sure how to test that since it will block, currently.

Reply:
Oh, could send it as a notification.